net/tcp/tcp_wrbuffer.c

```diff
@@ -152,17 +152,31 @@ uint32_t tcp_wrbuffer_inqueue_size(FAR struct tcp_conn_s *conn)
  *   of TCP data is about to sent
  *
  * Input Parameters:
- *   None
+ *   conn - The TCP connection of interest
  *
  * Assumptions:
  *   Called from user logic with the network locked.
  *
  ****************************************************************************/
 
-FAR struct tcp_wrbuffer_s *tcp_wrbuffer_alloc(void)
+FAR struct tcp_wrbuffer_s *tcp_wrbuffer_alloc(FAR struct tcp_conn_s *conn)
 {
   FAR struct tcp_wrbuffer_s *wrb;
 
+#if CONFIG_NET_SEND_BUFSIZE > 0
+  if (conn)
+    {
+      /* If the send buffer size exceeds the send limit, wait for the write
+       * buffer to be released
+       */
+
+      while (tcp_wrbuffer_inqueue_size(conn) > conn->snd_bufs)
+        {
+          net_lockedwait_uninterruptible(&conn->snd_sem);
```

Comment:
i guess this kind of things should be interruptible. but it isn't a new problem in this change.